### ISSUE TITLE
application freeze after interstitial.LoadAd(request) 

### ISSUE THREAD

**Issue by Lennotoecom** [googleads/googleads-mobile-unity#702]
Application freeze observed every time you run interstitial.LoadAd(request)

tested on the GoogleMobileAds version 3.13.1, 3.8.0, 3.7.0,
to reproduce the issue:
create empty project, import GoogleMobileAds, resolve dependencies,
code to trigger the issue:

```
using GoogleMobileAds;
using GoogleMobileAds.Api;
public class Stage : MonoBheavior {
	private InterstitialAd interstitial;
	private string adUnitId = "ca-app-pub-3940256099942544/1033173712";

	void Start() {
		private void requestIntersitial() {
			interstitial = new InterstitialAd (adUnitId);
			AdRequest request = new AdRequest.Builder ().Build ();
			interstitial.LoadAd (request);
		}
	}
}
```
Please help me this is a disaster.

**Comment by Pierre-Demessence** [comment]
Hello,

I get the same issue (v3.13.1):
After viewing an interstitial ad the app becomes totally irresponsive.
You're stuck and can't do anything but close it. Sometimes it would even crash directly.

There're several location where I load an intersitial ad but the issue doesn't occur everywhere.
It occurs only **when I try to load an image right after closing the ad** _(Loading some image raw data from the net or the disk, creating a texture from it and then a sprite from the texture)_.
Loading my image before showing the ad fixed the issue for me, but i'm pretty sure it shouldn't occur anyway.

I got a crash log of the issue with Android Studio attached to this post if it can helps.
[crashreport.txt](https://github.com/googleads/googleads-mobile-unity/files/2134242/crashreport.txt)

**Comment by Lennotoecom** [comment]
Hello colleagues,

thank you very much for your reply.
You can observe the issue on the following video which I recorded:
https://drive.google.com/open?id=1_qWUQ4xJd7pKc7vkO9R2_fmjvQmvxaF2

Code of the project:
```
using System.Collections;
using System.Collections.Generic;
using System;
using UnityEngine;
using UnityEngine.UI;
using GoogleMobileAds.Api;

public class Scene : MonoBehaviour {

	public GameObject counter;
	public GameObject timer;
	public GameObject log;

	private InterstitialAd interstitial;
	private string adUnitId = "ca-app-pub-3940256099942544/1033173712";

	void FixedUpdate () {
		timer.GetComponent<Text> ().text = " timer: " + ((int)Time.timeSinceLevelLoad).ToString ();
	}

	public void updateCounter(int value) {
		counter.transform.GetChild(0).GetComponent<Text> ().text = " " + value.ToString ();
		counter.GetComponent<Button>().onClick.RemoveAllListeners();
		counter.GetComponent<Button>().onClick.AddListener(() => updateCounter(value + 1));
	}

	public void requestIntersitial() {
		log.GetComponent<Text> ().text = "requestInterstitial::";
		interstitial = new InterstitialAd (adUnitId);
		interstitial.OnAdLoaded += HandleOnAdLoaded;
		interstitial.OnAdFailedToLoad += HandleOnAdFailedToLoad;
		AdRequest request = new AdRequest.Builder ().Build ();
		log.GetComponent<Text> ().text = "LoadAd";
		interstitial.LoadAd (request);
	}

	public void HandleOnAdLoaded(object sender, EventArgs args){
		log.GetComponent<Text> ().text = "HandleOnAdLoaded";
	}

	public void HandleOnAdFailedToLoad(object sender, EventArgs args){
		log.GetComponent<Text> ().text = "HandleOnAdFailedToLoad";
	}

}
```

The Project structure:
![projectelements](https://user-images.githubusercontent.com/5365155/41880664-672f7034-78ad-11e8-83b9-ce5bb6abc6a6.PNG)

ScreenView:
![sceneview](https://user-images.githubusercontent.com/5365155/41880681-76d9683c-78ad-11e8-94df-19e579d69745.PNG)

Camera element:
![cameraelement](https://user-images.githubusercontent.com/5365155/41880671-6da8da5e-78ad-11e8-9435-90b035c0a0b7.PNG)

requestAd element:
![requestadelement](https://user-images.githubusercontent.com/5365155/41880690-80e00e6c-78ad-11e8-8bab-a5b2d9f3d63f.PNG)

counter element:
![counterelement](https://user-images.githubusercontent.com/5365155/41880695-84c0cd28-78ad-11e8-9607-c95a892fb133.PNG)


On the video you can see that application stops responding for 3 seconds starting ~second 10 on timer.
On counter misbehavior seen after the number 13.

Thank you for investigating this.

**Comment by Lennotoecom** [comment]
Additional information:
issue observed only on Android devices (don't have apple),
I do not observe any errors or exceptions,
all consequent ad requests are running without freezes (only first one lags).
It appears that application itself is not frozen but input is impacted.

**Comment by Lennotoecom** [comment]
Hello colleagues,
please let me know if you have any update for this.

Thank you.

**Comment by Lennotoecom** [comment]
bump

**Comment by Lennotoecom** [comment]
bump

**Comment by rampara** [comment]
@Lennotoecom looks like the slowdown you're experiencing is due to GMA SDK initialization that occurs on the first ad request (and not something in this plugin). I have shared this issue the GMA SDK team and they are working towards refining initialization behavior. Closing this issue as it is not related to a bug in this plugin or a feature request, I would recommend you reach out on the [Google Mobile Ads Technical Support Forum](https://groups.google.com/forum/#!forum/google-admob-ads-sdk) if you would like to have further discussion on GMA SDK initialization and first request performance.
